exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -323,38 +717,32 @@ public AggOutcome doWork() {
                 if (EXTRA_DEBUG_1) {
                   logger.debug("Received new schema.  Batch has {} records.", incoming.getRecordCount());
                 }
-//                newSchema = true;
                 this.cleanup();
                 // TODO: new schema case needs to be handled appropriately
                 return AggOutcome.UPDATE_AGGREGATOR;
 
               case OK:
                 resetIndex();
-                if (incoming.getRecordCount() == 0) {
-                  continue;
-                } else {
-                  checkGroupAndAggrValues(currentIndex);
-                  incIndex();
-
-                  if (EXTRA_DEBUG_1) {
-                    logger.debug("Continuing outside loop");
-                  }
-                  continue outside;
+
+                if (EXTRA_DEBUG_1) {
+                  logger.debug("Continuing outside loop");
                 }
+                continue outside;
 
               case NONE:
-                // outcome = out;
+                underlyingIndex = 0; // in case need to handle a spilled partition
+                try { currentIndex = getVectorIndex(underlyingIndex); }
+                catch (SchemaChangeException sc) { throw new DrillRuntimeException(sc);}
```

Comment:
Done -- all SCE catchers either call UnsupportedOperationException, or (when clearly not possible then) IllegalStateException